### PR TITLE
BTHome V2

### PR DESCRIPTION
**Pull request by Ernst79**
PR to move the V2 documentation into main (Merge after release of V2)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -184,14 +184,16 @@ <h3>bthome-weather-station</h3>
           </div>
         </div>
 
-        <h3 id="example-data">Example Data</h3>
+        <h3 id="ble-advertising">BLE advertising in the BTHome V2 format</h3>
```
Comment (balloob) [inline index.html:187]: Given that this page is about the V2 format, should we reference it in the header/text?

File: index.html [hunk 2]
```diff
@@ -299,144 +316,67 @@ <h3 id="advertising-payload">Advertising payload</h3>
                   >After converting it to text</a
                 >
                 , it corresponds to "DIY-sensor". The name can be used to
-                identify your sensor.
+                identify your sensor. If you decide not to broadcast a name,
+                a default name "BTHome sensor" will be used. All names will
+                automatically get a postfix with the last 4 characters of the
+                MAC address to be able to identify different devices.
```
Comment (balloob) [inline index.html]: This is a consumer detail and there is no need to add this to the spec. If we want, we should change the text to be explicit that consumers need to implement it like that (and in that case I would drop "sensor" from name, as we also support buttons).

```suggestion
                identify your sensor.
```

File: index.html [hunk 3]
```diff
@@ -299,144 +316,67 @@ <h3 id="advertising-payload">Advertising payload</h3>
...
+          <li>bit 5-7: “BTHome Version”</li>
         </ul>
         <p>
-          Lets explain how the last two data packets work. The temperature
-          packet is used as example. The first byte
-          <code>0x23</code> (in bits <code>00100 011</code>) is giving
-          information about:
+          In the example, we have <code>0x40</code>.
+          After
+          <a href="https://www.mathsisfun.com/binary-decimal-hexadecimal-converter.html">converting</a>,
```
Comment (balloob) [inline index.html]:
```suggestion
          <a href="https://www.mathsisfun.com/binary-decimal-hexadecimal-converter.html">converting</a>
```

**Review by balloob — APPROVED** [review]

File: encryption.html [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        </code>
+        <p>
+          The result will be the decrypted data <code>02ca09 03bf13</code> like
+          we had on the top of this page.
+        </p>
+        <h3 id="examples">Example script</h3>
+        <p>
+          A detailed
+          <a
+            href="https://github.com/Bluetooth-Devices/bthome-ble/blob/main/src/bthome_ble/bthome_v2_encryption.py"
```
Comment (kzyapkov) [inline encryption.html:206]: Dead link?

Comment (Ernst79) [reply encryption.html:206]: The links should be correct after the release.

File: index.html (hunk 1 above)
Comment (Ernst79) [reply index.html:187]: Added V2 in the first title and the text below.

File: index.html (hunk 2 above)
Comment (Ernst79) [reply index.html]: agreed

File: index.html (hunk 3 above)
Comment (Ernst79) [reply index.html]: fixed

**Pull request merged**